doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,971 @@
+```
+
+## 3. CLI Commands for CMIS Diagnostic Monitoring
+
+### 3.1 CLI Commands for DOM Monitoring
+
+#### 3.1.1 `show interfaces transceiver dom PORT`
+
+This CLI shows the transceiver DOM and threshold values for a given port.
+
+```plaintext
+CLI output format:
+                              High Alarm   High Warning   Low Warning   Low Alarm
+             Paramter_Name    Threshold    Threshold      Threshold     Threshold
+Port         (Unit)           (Unit)       (Unit)         (Unit)        (Unit)
+-----------  ---------------  --------     --------       --------      --------
+
+Example:
+admin@sonic#show interfaces transceiver dom Ethernet1
+                              High Alarm   High Warning   Low Warning   Low Alarm
+             Temperature      Threshold    Threshold      Threshold     Threshold
+Port         (Celsius)        (Celsius)    (Celsius)      (Celsius)     (Celsius)
+-----------  ---------------  --------     --------       --------      --------
+Ethernet1    100              90           80             -10            -20
+```
+
+#### 3.1.2 `show interfaces transceiver dom flag PORT`
+
+This CLI shows the transceiver DOM flags for a given port.
+
+```plaintext
+CLI output format:
+                              High Alarm   High Warning   Low Warning   Low Alarm
+Port         Paramter_Name    Flag         Flag           Flag          Flag
+-----------  ---------------  --------     --------       --------      --------
+
+Example:
+admin@sonic#show interfaces transceiver dom flag Ethernet1
+                              High Alarm   High Warning   Low Warning   Low Alarm
+Port         Paramter_Name    Flag         Flag           Flag          Flag
+-----------  ---------------  --------     --------       --------      --------
+Ethernet1    Temperature      False        False          False         False
+```
+
+### 3.2 CLI Commands for VDM Monitoring
+
+#### 3.2.1 `show interfaces transceiver vdm PORT`
+
+This CLI shows the transceiver VDM and threshold values for a given port.
+
+```plaintext
+CLI output format:
+                              High Alarm   High Warning   Low Warning   Low Alarm
+             Paramter_Name    Threshold    Threshold      Threshold     Threshold
+Port         (Unit)           (Unit)       (Unit)         (Unit)        (Unit)
+-----------  ---------------  --------     --------       --------      --------
+
+Example:
+admin@sonic#show interfaces transceiver vdm Ethernet1
+Basic Values:
+                              High Alarm   High Warning   Low Warning   Low Alarm
+             eSNR Media Input Threshold    Threshold      Threshold     Threshold
+Port         (dB)             (dB)         (dB)           (dB)          (dB)
+-----------  ---------------  --------     --------       --------      --------
+Ethernet1    23.48046875      0            0              0             0
+```
+
+#### 3.2.2 `show interfaces transceiver vdm flag PORT`
```

Comment:
@mihirpat1 would need to capture these in sonic-utilities as well for user guide